extensions/lifecycle/bucketProcessor/LifecycleBucketProcessor.js

```diff
@@ -76,6 +77,17 @@ class LifecycleBucketProcessor {
         this._producer = null;
         this._kafkaBacklogMetrics = null;
 
+        this._supportedRulesObject = {};
+        supportedLifecycleRules.forEach(rule => {
+            this._supportedRulesObject[rule] = { enabled: true };
+        });
+        this._hasSupportedRules = Array.isArray(supportedLifecycleRules) &&
+            supportedLifecycleRules.length > 0;
```

Comment:
If `supportedLifecycleRules` can be something other than an array I think we should guard the `forEach` as well
```suggestion
        this._hasSupportedRules = Array.isArray(supportedLifecycleRules) &&
            supportedLifecycleRules.length > 0;
        this._supportedRulesObject = {};
        if (this._hasSupportedRules) {
            supportedLifecycleRules.forEach(rule => {
                this._supportedRulesObject[rule] = { enabled: true };
            });
        }
```

Reply:
`supportedLifecycleRules` is a constant from [arsenal](https://github.com/scality/Arsenal/blob/development/8.1/lib/constants.js#L146); so it safe to say that it is always an array.